src/components/datePicker/datePicker.tsx

```diff
@@ -0,0 +1,126 @@
+import ReactDatePicker, { registerLocale } from 'react-datepicker';
+import classNames from 'classnames/bind';
+import { Locale } from 'date-fns';
+import { FC, ReactNode, useRef, ReactElement } from 'react';
+import FieldText from '@components/fieldText';
+import { CalendarIcon } from '@/components/icons';
+import { DatePickerHeader } from './header/datePickerHeader';
+import styles from './datePicker.module.scss';
+
+const cx = classNames.bind(styles);
+
+const DEFAULT_LANGUAGE = 'en';
+const DEFAULT_DATE_FORMAT = 'MM-dd-yyyy';
+
+interface DatePickerProps {
+  onChange?: (date: Date | any) => void;
+  onBlur?: () => void;
+  onFocus?: () => void;
+  headerNodes?: ReactNode;
+  disabled?: boolean;
+  shouldCloseOnSelect?: boolean;
+  fixedHeight?: boolean;
+  startDate?: Date | undefined;
+  endDate?: Date | undefined;
+  customClassName?: string;
+  popperClassName?: string;
+  calendarClassName?: string;
+  customTimeInput?: ReactElement;
+  language?: string;
+  yearsOptions?: number[];
+  placeholder?: string;
+  dateFormat?: string;
+  selects?: 'start' | 'end' | string;
+  value?: Date | null;
+}
+
+export const registerDatePickerLocale = (language: string, locale: Locale) => {
+  registerLocale(language, locale);
+};
+
+export const DatePicker: FC<DatePickerProps> = ({
+  onChange = () => {},
+  disabled = false,
+  onBlur = () => {},
+  onFocus = () => {},
+  endDate = undefined,
+  startDate = undefined,
+  headerNodes = null,
+  customClassName = '',
+  customTimeInput = undefined,
+  shouldCloseOnSelect = true,
+  popperClassName = '',
+  calendarClassName = '',
+  fixedHeight = false,
+  language = DEFAULT_LANGUAGE,
+  yearsOptions = [],
+  placeholder = DEFAULT_DATE_FORMAT.toUpperCase(),
+  dateFormat = DEFAULT_DATE_FORMAT,
+  selects = '',
+  value = null,
+}) => {
+  const inputRef = useRef(null);
+  const startDateToString = startDate?.toDateString();
```

Comment:
Sounds more like a function name, may be `startDateString`?
Check other similar cases as well.